qiskit/aqua/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -251,55 +278,16 @@ def print_settings(self):
         ret += "===============================================================\n"
         return ret
 
-    def _config_the_best_mode(self, operator, backend):
-        if not isinstance(operator, (WeightedPauliOperator, MatrixOperator,
-                                     TPBGroupedWeightedPauliOperator)):
-            logger.debug("Unrecognized operator type, skip auto conversion.")
-            return operator
-
-        ret_op = operator
-        if not is_statevector_backend(backend) and not (
-                is_aer_provider(backend)
-                and self._quantum_instance.run_config.shots == 1):
-            if isinstance(operator, (WeightedPauliOperator, MatrixOperator)):
-                logger.debug("When running with Qasm simulator, grouped pauli can "
-                             "save number of measurements. "
-                             "We convert the operator into grouped ones.")
-                ret_op = op_converter.to_tpb_grouped_weighted_pauli_operator(
-                    operator, TPBGroupedWeightedPauliOperator.sorted_grouping)
-        else:
-            if not is_aer_provider(backend):
-                if not isinstance(operator, MatrixOperator):
-                    logger.info("When running with non-Aer statevector simulator, "
-                                "represent operator as a matrix could "
-                                "achieve the better performance. We convert "
-                                "the operator to matrix.")
-                    ret_op = op_converter.to_matrix_operator(operator)
-            else:
-                if not isinstance(operator, WeightedPauliOperator):
-                    logger.info("When running with Aer simulator, "
-                                "represent operator as weighted paulis could "
-                                "achieve the better performance. We convert "
-                                "the operator to weighted paulis.")
-                    ret_op = op_converter.to_weighted_pauli_operator(operator)
-        return ret_op
-
-    def construct_circuit(self, parameter: Union[List[float], List[Parameter], np.ndarray],
-                          statevector_mode: bool = False,
-                          use_simulator_snapshot_mode: bool = False,
-                          circuit_name_prefix: str = '') -> List[QuantumCircuit]:
-        """Generate the circuits.
+    def construct_circuit(self,
```

Comment:
I can do either - I actually originally removed this method because it's a bit ambiguous (we generate many circuits for the expectation value), but added it back after Julien's Native Circuits PR and used it to just hold Julien's logic to handle both circuits and VarForms. Should I remove it, or make it return a list of circuits for one expectation, or the expect_op, which will include both the circuits and the expectation logic for a single expectation evaluation?

Reply:
People had used this to both look at the circuit and run it themselves. In taking parameters it would configure the circuits to compute the expectation value with the given set of params and return that. All the algorithms have this ability to pass back the circuit they run so it can be inspected, run etc. It would be good to continue this.

Reply:
Ok - so in the new flow this would be returning expectation_op that holds the circuits before sampling and the measurement. That would also be better because they can both run it and print it to look at the circuits. Should I do that? If I return a list of circuits to the user they still don't have what they'd need to compute the expectation out of them, as we don't have the old Operator's function for doing that. It would basically just be for printing, which they can do with the expectation_op anyway.

Reply:
It would be whatever vqe itself runs for each computation is that's possible. Users wanted to be able to see the circuits VQE (or any other algo) ran and be able to run them themselves.

Reply:
Ok - so that's the expectation_op. I'll change to return that.

Reply:
Ok, fixed